Design task/task 2 updated

Conormc98 opened the pull request.

JamesMatchett commented on ExpressAPI/index.js:
```diff
@@ -11,7 +11,7 @@ var options = { cookies: true }
 var jwt = require('express-jwt');
 
-app.use(jwt({ secret: process.env.JWT_Secret}).unless({path: ['/login', '/SecureGenerateUser']}));
+app.use(jwt({ secret: process.env.JWT_Secret}).unless({path: ['/login', '/SecureGenerateUser', '/getBandRoles/4']}));
```
```suggestion
app.use(jwt({ secret: process.env.JWT_Secret}).unless({path: ['/login', '/SecureGenerateUser']}));
```

JamesMatchett replied:
Remove getBand from exclusion list

ChrisThompsonK reviewed (approved):
These changes look good to me 👍

micalexis commented on ExpressAPI/index.js:
```diff
@@ -174,15 +174,23 @@ app.get('/InvalidateUserToken', function(req, res){
-})
+});
 
 app.get('/IsUserValid', function(req,res){
```
refactor your endpoints to make them RESTful

micalexis replied:
?

micalexis commented on kjourney/src/app/bands/bands.component.html:
```diff
@@ -1,68 +1,91 @@
+<header>
+  <br>
+  <br>
+  <h1 style="margin-top: 20px; text-align: center;" *ngFor="let band of bands">{{ band.bandName }}</h1>
```
why is this styling for the template not in the .css for the component?

micalexis commented on kjourney/src/app/bands/bands.component.html:
```diff
@@ -1,68 +1,91 @@
-  <a href="#">Support</a>
-</div>
+<header>
+  <br>
```
cant this be achieved with styling?

micalexis reviewed (changes requested):
address comments

Commit:
'Improved' data - all link tables filled for 6 complete job families, ft. chrises removal of silly things

Added training link field
commits squashed

Created new branch, and moved CSS from band into bands

Added Query Params for Bands page

Removed 'band'

Fixed styling for Bands page

Update ExpressAPI/index.js

Co-Authored-By: James Matchett <30292929+JamesMatchett@users.noreply.github.com>

Added changes from code review

Added dividers between Competencies

Address code review changes

Made API RESTful

micalexis commented on kjourney/src/app/bands/bands.component.html:
```diff
@@ -1,68 +1,89 @@
       </tr>
       </thead>
       <tbody>
       <tr class="myRow" id="pop-over" *ngFor="let trainingPerBands of trainingPerBand">
```
how can you get a plural from a singular?
change the names of the variables to make sense

micalexis commented on kjourney/src/app/data.service.ts:
```diff
@@ -28,4 +30,12 @@ export class DataService {
   }
+
+  public getBandRoles(bandID): Observable<BandRole[]> {
+    return this.http.get<BandRole[]>('api/roles/' + bandID);
```
you are getting roles by bandId so you need to use query parameters 

`api/roles/?bandId=`

micalexis reviewed (changes requested):
address comments and squash commits

micalexis reviewed (approved):
squash commits
